### PR TITLE
Fix stable diffusion export

### PR DESCRIPTION
**Pull request by JingyaHuang**
Fix bugs introduced by the update of the optimum main repo.

### CODE REVIEW

**Review by philschmid — APPROVED** [review]
LGTM

**Review by dacorvo — APPROVED** [review]

**Comment by dacorvo** [comment]
In addition to the usual errors we have, there seems to be an issue with stable diffusion export tests: `'GPT2NeuronConfig' has no attribute 'get_mandatory_axes_for_task'`.

**Comment by JingyaHuang** [comment]
> In addition to the usual errors we have, there seems to be an issue with stable diffusion export tests: `'GPT2NeuronConfig' has no attribute 'get_mandatory_axes_for_task'`.

@dacorvo, it's interesting as `GPT2NeuronConfig` is not supposed to be used during the sd test 🤔.

Could you give me a snippet so that I can reproduce it? Thanks!

**Comment by dacorvo** [comment]
> > In addition to the usual errors we have, there seems to be an issue with stable diffusion export tests: `'GPT2NeuronConfig' has no attribute 'get_mandatory_axes_for_task'`.
> 
> @dacorvo, it's interesting as `GPT2NeuronConfig` is not supposed to be used during the sd test 🤔.
> 
> Could you give me a snippet so that I can reproduce it? Thanks!

Yes, my bad, this error is for the next test. The stable diffusion error is about a string passed as a list index.
This is the log from the CI: https://github.com/huggingface/optimum-neuron/actions/runs/5598435281/jobs/10238146802?pr=144.

File: optimum/exporters/neuron/model_configs.py [hunk 1]
```diff
@@ -179,28 +179,49 @@ def outputs(self) -> List[str]:
...
     NORMALIZED_CONFIG_CLASS = NormalizedConfig.with_args(
         vocab_size="vocab_size",
         sequence_length="max_position_embeddings",
+        num_layers="num_hidden_layers",
         allow_new=True,
     )
 
     @property
-    def inputs(self) -> List[str]:
+    def inputs(self) -> Dict[str, Dict[int, str]]:
```
Comment (michaelbenayoun) [inline optimum/exporters/neuron/model_configs.py:194]: ?

File: optimum/exporters/neuron/model_configs.py [hunk 2]
```diff
@@ -179,28 +179,49 @@ def outputs(self) -> List[str]:
...
         return ["input_ids"]
 
+    @property
+    def outputs(self) -> Dict[str, Dict[int, str]]:
```
Comment (michaelbenayoun) [inline optimum/exporters/neuron/model_configs.py:198]:
```suggestion
    def outputs(self) -> Dict[str, Dict[int, str]]:
```
?

File: optimum/exporters/neuron/model_configs.py [hunk 3]
```diff
@@ -179,28 +179,49 @@ def outputs(self) -> List[str]:
...
+        if self._normalized_config.output_hidden_states:
+            for i in range(self._normalized_config.num_layers + 1):
+                common_outputs.append(f"hidden_states.{i}")
+
+        return common_outputs
 
     def generate_dummy_inputs(self, return_tuple: bool = False, **kwargs):
         dummy_inputs = super().generate_dummy_inputs(**kwargs)
-        dummy_inputs["input_ids"] = dummy_inputs["input_ids"].to(dtype=torch.int32)
+        dummy_inputs["input_ids"] = dummy_inputs["input_ids"]  # .to(dtype=torch.int32)
```
Comment (michaelbenayoun) [inline optimum/exporters/neuron/model_configs.py:224]: To remove?

File: optimum/exporters/neuron/model_configs.py [hunk 4]
```diff
@@ -211,6 +232,52 @@ def check_model_inputs_order(self, model, dummy_inputs, forward_with_tuple=False
...
+    @property
+    def outputs(self) -> List[str]:
+        return ["sample"]
+
+    def generate_dummy_inputs(self, return_tuple: bool = False, **kwargs):
+        dummy_inputs = super().generate_dummy_inputs(**kwargs)
+        dummy_inputs["timestep"] = dummy_inputs["timestep"].float()
+        dummy_inputs["encoder_hidden_states"] = dummy_inputs["encoder_hidden_states"][0]
+
+        if return_tuple is True:
```
Comment (michaelbenayoun) [inline optimum/exporters/neuron/model_configs.py:275]:
```suggestion
        if return_tuple:
```

**Pull request merged**